cmd/bootstrap/run/execution_state.go

```diff
@@ -36,10 +36,10 @@ func GenerateExecutionState(
 	metricsCollector := &metrics.NoopCollector{}
 
 	ledgerStorage, err := ledger.NewLedger(dbDir, 100, metricsCollector, zerolog.Nop(), nil)
-	defer ledgerStorage.CloseStorage()
 	if err != nil {
 		return nil, err
 	}
+	defer ledgerStorage.CloseStorage()
```

Comment:
why moving this?

Reply:
Because if there is error `ledgerStorage` is `nil` and will fail with closing.